projects/storefrontlib/src/layout/a11y/keyboard-focus/visible/visible-focus.directive.ts

```diff
@@ -23,13 +23,29 @@ export class VisibleFocusDirective extends BaseFocusDirective {
     }
   }
 
-  @HostListener('keydown') handleKeydown() {
+  @HostListener('keydown', ['$event']) handleKeydown(event: KeyboardEvent) {
     if (this.shouldFocusVisible) {
-      this.mouseFocus = false;
+      this.mouseFocus = !this.isMouseEvent(event);
     }
   }
 
   protected get shouldFocusVisible(): boolean {
     return this.config?.disableMouseFocus;
   }
+
+  /**
+   * Indicates whether the current event is driven by the mouse or the keyboard.
+   *
+   * If the keyboard is used to navigate the OS or the browser, or used ot fill in a form,
+   * we consider the event to be a mouse event.
+   */
+  protected isMouseEvent(event: KeyboardEvent): boolean {
+    return (
+      event.code === 'Tab' ||
+      !(
+        event.metaKey ||
+        (this.mouseFocus && (event.target as HTMLElement).tagName === 'INPUT')
```

Comment:
I'm not sure I understand why setting the new value for `this.mouseFocus` depends on the previous value of the `this.mouseFocus`

Reply:
This is needed to distinquish the keyboard user from mouse user. The keyboard user will keep the focus styling, where as the keyboard user won't.